python/hopsworks/connection.py

```diff
@@ -239,12 +241,33 @@ def connect(self):
         try:
             # init client
             if client.base.Client.REST_ENDPOINT not in os.environ:
+                # determine engine, needed to init client
```

Comment:
One can see it as a part of future work to unite connections. Technically, it is needed at the moment, since I replaced `_hsfs_post_init` with (hopefully) a bit more understandable `provide_project`. The new function gets engine from `_engine` of client, which has to be set up in initializer.

In the end, after doing hopsworks.login the user should be able to use all the functionality of hopsworks, including hsfs-related.